### PR TITLE
[RCCA-17800] Fix `confluent kafka cluster configuration update` config length bug

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- Fix an issue in `confluent kafka cluster configuration update` causing some valid inputs to return a duplicate key error

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----
When updating a configuration that accepts a list as a value, cobra will split that input along commas. When we parse this into a map we get what we intend.

But we make `[]AlterConfigBatchRequestDataData` using the length of the input slice instead of the length of the map. This causes empty key/value pairs to be sent to the backend. If there are at least 2 of these, the backend returns an error over duplicate keys.

The fix is simply to use the length of the map and not the slice when creating the request data.

References
----------

Test & Review
-------------
Manual testing:
```
confluent kafka cluster configuration update --config ssl.cipher.suites="TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256"
Successfully requested to update configuration "ssl.cipher.suites".
```
```
confluent kafka cluster configuration update --config ssl.cipher.suites="TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",auto.create.topics.enable=true
Successfully requested to update configurations "auto.create.topics.enable" and "ssl.cipher.suites".
```

Also added a check in the testing backend to return a Bad Request if it receives duplicate keys, and added a corresponding new integration test. I checked that the test succeeds with this change but fails without it.

### CODE REVIEW

File: test/kafka_test.go [hunk 1]
```diff
@@ -172,6 +172,7 @@ func (s *CLITestSuite) TestKafkaClusterConfiguration() {
 		{args: "kafka cluster use lkc-12345"},
 		{args: "kafka cluster configuration describe compression.type", fixture: "kafka/cluster/configuration/describe.golden"},
 		{args: "kafka cluster configuration update --config auto.create.topics.enable=true", fixture: "kafka/cluster/configuration/update.golden"},
+		{args: "kafka cluster configuration update --config ssl.cipher.suites=\"val1,val2,val3\"", fixture: "kafka/cluster/configuration/update-with-commas.golden"},
```
Comment (brianstrauch) [inline test/kafka_test.go:175]:
```suggestion
		{args: `kafka cluster configuration update --config ssl.cipher.suites="val1,val2,val3"`, fixture: "kafka/cluster/configuration/update-with-commas.golden"},
```

File: test/test-server/kafka_rest_router.go [hunk 2]
```diff
@@ -1791,8 +1791,22 @@ func handleKafkaBrokerIdConfigs(t *testing.T) http.HandlerFunc {
 }
 
 // Handler for: "/kafka/v3/clusters/{cluster_id}/broker-configs:alter"
-func handleKafkaBrokerConfigsAlter(_ *testing.T) http.HandlerFunc {
+func handleKafkaBrokerConfigsAlter(t *testing.T) http.HandlerFunc {
```
Comment (brianstrauch) [inline test/test-server/kafka_rest_router.go:1794]: Looks like we're still not using this
```suggestion
func handleKafkaBrokerConfigsAlter(_ *testing.T) http.HandlerFunc {
```

File: test/test-server/kafka_rest_router.go [hunk 3]
```diff
@@ -1791,8 +1791,22 @@ func handleKafkaBrokerIdConfigs(t *testing.T) http.HandlerFunc {
...
+
+			dataMap := make(map[cckafkarestv3.AlterConfigBatchRequestDataData]bool)
+			for _, data := range req.Data {
+				dataMap[data] = true
+			}
+
+			if len(dataMap) < len(req.Data) {
+				w.WriteHeader(http.StatusBadRequest)
+			}
+			return
```
Comment (brianstrauch) [inline test/test-server/kafka_rest_router.go:1808]:
```suggestion
```

File: test/test-server/kafka_rest_router.go [hunk 4]
```diff
@@ -1791,8 +1791,22 @@ func handleKafkaBrokerIdConfigs(t *testing.T) http.HandlerFunc {
...
+func handleKafkaBrokerConfigsAlter(t *testing.T) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost {
+			var req cckafkarestv3.AlterConfigBatchRequestData
+			_ = json.NewDecoder(r.Body).Decode(&req)
+
+			dataMap := make(map[cckafkarestv3.AlterConfigBatchRequestDataData]bool)
+			for _, data := range req.Data {
+				dataMap[data] = true
+			}
```
Comment (brianstrauch) [inline test/test-server/kafka_rest_router.go:1803]: I think we have a package that can convert a slice to a set

**Review by brianstrauch — COMMENTED** [review]
Thank you! (This bug was 100% my fault)

File: test/test-server/kafka_rest_router.go (hunk 4 above)
Comment (sgagniere) [reply test/test-server/kafka_rest_router.go:1803]: Yeah I tried it but this type isn't `comparable`

File: test/test-server/kafka_rest_router.go (hunk 2 above)
Comment (sgagniere) [reply test/test-server/kafka_rest_router.go:1794]: Oops. At one point I thought I needed it, and I forgot to change it back when I realized I didn't.

File: internal/kafka/command_cluster_configuration_update.go [hunk 5]
```diff
@@ -54,7 +54,7 @@ func (c *clusterCommand) configurationUpdate(cmd *cobra.Command, _ []string) err
 		return err
 	}
 
-	data := make([]kafkarestv3.AlterConfigBatchRequestDataData, len(config))
+	data := make([]kafkarestv3.AlterConfigBatchRequestDataData, len(configMap))
```
[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/kafka/command_cluster_configuration_update.go:57]: I'll take a look

[the remark this answers is not in the document]
Comment (sgagniere) [reply internal/kafka/command_cluster_configuration_update.go:57]: Doesn't look like it.

**Review by brianstrauch — APPROVED** [review]
